### PR TITLE
Add question 9: Other kinds of support

### PR DESCRIPTION
**Pull request by huwd**
Trello: https://trello.com/c/AC88YKom/30-implement-question-9

# What
Adds qusetion 9, text area for other support.

# Screenshots
image

image

### CODE REVIEW

File: app/controllers/coronavirus_form/offer_other_support_controller.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    end
+  end
+
+private
+
+  PAGE = "offer_other_support"
+  NEXT_PAGE = "your_business_details"
+
+  def previous_path
+    "/"
```
Comment (bilbof) [inline app/controllers/coronavirus_form/offer_other_support_controller.rb]: Just need to update this bit

File: app/controllers/coronavirus_form/offer_other_support_controller.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      PAGE,
+      other: support_other,
+    )
+
+    if invalid_fields.any?
+      flash.now[:validation] = invalid_fields
+      render "coronavirus_form/#{PAGE}"
+    elsif session["check_answers_seen"]
+      redirect_to controller: "coronavirus_form/check_answers", action: "show"
+    elsif session[:medical_equipment] == I18n.t("coronavirus_form.medical_equipment.options.option_yes.label")
```
Comment (bilbof) [inline app/controllers/coronavirus_form/offer_other_support_controller.rb]: Does `medical_equipment` need to be `support_other` or could the line be removed?

Comment (huwd) [reply app/controllers/coronavirus_form/offer_other_support_controller.rb]: yeah sorry, just put this up to have somewhere to park it, still [WIP]!

File: app/controllers/coronavirus_form/offer_other_support_controller.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      PAGE,
+      other: support_other,
+    )
+
+    if invalid_fields.any?
+      flash.now[:validation] = invalid_fields
+      render "coronavirus_form/#{PAGE}"
+    elsif session["check_answers_seen"]
+      redirect_to controller: "coronavirus_form/check_answers", action: "show"
+    elsif session[:offer_other_support] == I18n.t("coronavirus_form.offer_other_support.options.option_yes.label")
```
Comment (bilbof) [inline app/controllers/coronavirus_form/offer_other_support_controller.rb]: It looks like this line would need to be updated as we're missing an `option_yes.label` for the section.

**Review by bilbof — COMMENTED** [review]
Just a minor comment.

Please could we have a test for this page?

I wonder if we could also set a character limit on the textarea (not just this one), since we're sticking this in a cookie for now which (iirc) has a 4kb limit.

**Comment by huwd** [comment]
Yes sorry, should have been there to begin with!

File: app/controllers/coronavirus_form/offer_other_support_controller.rb (hunk 3 above)
Comment (huwd) [reply app/controllers/coronavirus_form/offer_other_support_controller.rb]: great added that, thanks for the spot

File: app/controllers/coronavirus_form/offer_other_support_controller.rb [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    end
+  end
+
+private
+
+  PAGE = "offer_other_support"
+  NEXT_PAGE = "thank_you"
+
+  def previous_path
+    "/"
```
Comment (bilbof) [inline app/controllers/coronavirus_form/offer_other_support_controller.rb]:
```suggestion
    coronavirus_form_offer_community_support_path
```

**Review by bilbof — APPROVED** [review]
Minor change required, but otherwise good to go 👍 fab work

File: app/views/coronavirus_form/offer_other_support.html.erb [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+<% end %>
+
+<%= render "coronavirus_form/validation_error" %>
+<%= form_tag({},
+  "data-module": "track-coronavirus-form-offer-other-support",
+  "data-question-key": "offer_other_support",
+  "id": "offer_other_support"
+) do %>
+
+<%# Terrible hack for now, will move this into the component properly before launch %>
```
Comment (alex-ju) [inline app/views/coronavirus_form/offer_other_support.html.erb]: Updating `govuk_publishing_component` to 21.32.0 will provide support for this use case. Example here: https://components.publishing.service.gov.uk/component-guide/textarea/with_label_as_page_heading.

File: app/controllers/coronavirus_form/offer_other_support_controller.rb (hunk 4 above)
Comment (huwd) [reply app/controllers/coronavirus_form/offer_other_support_controller.rb]: thanks very much, resolved!

File: app/views/coronavirus_form/offer_other_support.html.erb (hunk 5 above)
Comment (huwd) [reply app/views/coronavirus_form/offer_other_support.html.erb]: thanks very much for that alex, have amended it!

**Pull request merged**